### ISSUE TITLE
Xorriso Failure on Antergos Minimal ISO

### ISSUE THREAD

**Issue by ngustavo** [emerinohdz/arch64-efi32#1]
I'm trying to run your script on Antergos Minimal and Manjaro Architect ISOs. I'm not sure why it's not working. Could you help me, please?
```
xorriso 1.4.2 : RockRidge filesystem manipulator, libburnia project.

Drive current: -outdev 'stdio:target/arch-efi32.iso'
Media current: stdio file, overwriteable
Media status : is blank
Media summary: 0 sessions, 0 data blocks, 0 data,  225g free
Added to ISO image: directory '/'='/home/gustavo/Downloads/iso/arch-ia32/arch64-efi32-master/target/iso'
xorriso : FAILURE : Given path does not exist on disk: -boot_image system_area='/home/gustavo/Downloads/iso/arch-ia32/arch64-efi32-master/target/iso/isolinux/isohdpfx.bin'
xorriso : FAILURE : Cannot find in ISO image: -boot_image ... bin_path='/isolinux/isolinux.bin'
xorriso : aborting : -abort_on 'FAILURE' encountered 'FAILURE'
[10:57:03.076] Task 'dist' failed after 5 ms (5)
[10:57:03.079] Task 'default' failed after 55 ms (5)
gustavo@desk ~/Downloads/iso/arch-ia32/arch64-efi32-master $ 
```

**Comment by emerinohdz** [comment]
Hello, take a look at these lines:

xorriso : FAILURE : Given path does not exist on disk: -boot_image system_area='/home/gustavo/Downloads/iso/arch-ia32/arch64-efi32-master/target/iso/isolinux/isohdpfx.bin'
xorriso : FAILURE : Cannot find in ISO image: -boot_image ... bin_path='/isolinux/isolinux.bin'

It seems xorriso cannot find the iso, from your comment I take it you are not using an archlinux ISO, but you should download the ISO of the distro you are using to the chosen path, take a look at the script file so you can modifiy the path in case you need to.

Please let me know if you figure it out.

**Comment by emerinohdz** [comment]
Please, check the runnerfile.sh file, it has the default location it expects for the ISO image, let me know if you need any help.

**Comment by ngustavo** [comment]
Thank you. I did that but it seems that Manjaro and Antergos don't have the isolinux folder nor isohdpfx.bin. I tried with archlinux then, but it throws me this error:
`grub-mkstandalone: error: /usr/lib/grub/i386-efi/modinfo.sh doesn't exist. Please specify --target or --directory.`
I'm running it on Linux Mint and I've checked that file. It does exist.

**Comment by emerinohdz** [comment]
It seems you are missing grub for i386 on that OS, verify that you have it or download it if it's missing. That is needed in order to generate the correct ISO.

**Comment by ngustavo** [comment]
So it worked on Arch. Thank you.
I found a xorriso command in manjaro-tools and tweaked it for your script:
```
runner_run xorriso -as mkisofs \
        --protective-msdos-label \
        -volid "MANJARO_EFI32" \
        -r -graft-points -no-pad \
        --sort-weight 0 / \
        --sort-weight 1 /boot \
        --grub2-mbr target/iso/boot/grub/i386-pc/boot_hybrid.img \
        -partition_offset 16 \
        -b boot/grub/i386-pc/eltorito.img \
        -c boot.catalog \
        -no-emul-boot -boot-load-size 4 -boot-info-table --grub2-boot-info \
        -eltorito-alt-boot \
        -append_partition 2 0xef ./target/iso/efi.img \
        -e --interval:appended_partition_2:all:: \
        -no-emul-boot \
        -iso-level 3 \
        -o target/manjaro-efi32.iso \
        target/iso
```
But it throws this error:
`xorriso : FAILURE : Cannot find path '/--interval:appended_partition_2:all::' in loaded ISO image`
I guess xorriso is interpreting that as a regular file path instead of a flag.

**Comment by emerinohdz** [comment]
Great it worked. However, it'll be good to know the version you used in Manjaro for xorriso, so I know which xorriso version is not compatible with this script.

**Comment by ngustavo** [comment]
I'm not sure if that's the case but this is it.
`$ xorriso -version`
```
xorriso 1.4.2 : RockRidge filesystem manipulator, libburnia project.

xorriso 1.4.2
ISO 9660 Rock Ridge filesystem manipulator and CD/DVD/BD burn program
Copyright (C) 2015, Thomas Schmitt <scdbackup@gmx.net>, libburnia project.
xorriso version   :  1.4.2
Version timestamp :  2015.11.28.140001
Build timestamp   :  -none-given-
libisofs   in use :  1.4.2  (min. 1.4.2)
libjte     in use :  1.0.0  (min. 1.0.0)
libburn    in use :  1.4.2  (min. 1.4.2)
libburn OS adapter:  internal GNU/Linux SG_IO adapter sg-linux
libisoburn in use :  1.4.2  (min. 1.4.2)
Provided under GNU GPL version 3 or later, due to libreadline license.
There is NO WARRANTY, to the extent permitted by law.
```

**Comment by emerinohdz** [comment]
Hello, I checked your xorriso command, it is throwing that error because the "-e" option expects the path to the efiboot image that contains the bootia32.efi binary (created in task_build), so try changing this line:

> -e --interval:appended_partition_2:all:: 

to this:

`-e EFI/archiso/efiboot.img \`
`--interval:appended_partition_2:all:: \`

Let me know if that works.

**Comment by emerinohdz** [comment]
I checked the documentation for xorriso and the -e option, it seems what you are doing should work, maybe it wasn't implemented on v1.4.2?

**Comment by ngustavo** [comment]
Could be the case. I can't run a newer version on Mint 18.2. Maybe when I have more time I can work on it from Manjaro Live. Turns out that Ubuntu uses almost the same xorriso command as Arch.
I tried it on Arch Anywhere also. It has some weird `[BOOT]` folder. I'm not a xorriso expert so I left it but it gives me a blank screen. Probably not finding the kernel.

Thank you for helping me. I'll choose between Arch and Ubuntu, maybe KDE since it's a tablet.
